core/src/services/s3/core.rs

```diff
@@ -462,7 +463,7 @@ impl S3Core {
         // Set user metadata headers.
         if let Some(user_metadata) = args.user_metadata() {
             for (key, value) in user_metadata {
-                req = req.header(format!("{}{}", constants::X_AMZ_META_PREFIX, key), value)
+                req = req.header(format!("{X_AMZ_META_PREFIX}{key}"), value)
```

Comment:
Changed this in order to be consistent with what I did in the azure part. It is more idiomatic to use string interpolation